common/config/config.go

```diff
@@ -742,3 +742,66 @@ func (k *KeepAliveServerConfig) GetKeepAliveEnforcementPolicy() keepalive.Enforc
 
 	return defaultConfig
 }
+
+func (fi *FaultInjection) WithError(storeName DataStoreName, methodName, errorName string, probability float64) *FaultInjection {
+	if fi == nil {
+		return nil
+	}
+	m := fi.method(storeName, methodName)
+	m.Errors[errorName] = probability
+	fi.Targets.DataStores[storeName].Methods[methodName] = m
```

Comment:
should we use pointers for FaultInjectionMethodConfig so this can be
```suggestion
	fi.method(storeName, methodName).Errors[errorName] = probability
```
?
kinda weird for `method` to set up maps but then return a copied value

Reply:
I gave it a quick look. It will require fixes on caller side which is autogenerated. I might do it separately.